Userland/Libraries/LibGfx/Size.h

```diff
@@ -87,6 +87,29 @@ class Size {
         return size;
     }
 
+    [[nodiscard]] constexpr float aspect_ratio() const
+    {
+        return static_cast<float>(width()) / static_cast<float>(height());
+    }
+
+    // Horizontal means preserve the width, Vertical means preserve the height.
+    [[nodiscard]] constexpr Size<T> match_aspect_ratio(float aspect_ratio, Orientation side_to_preserve) const
+    {
+        auto matched = *this;
+        auto height_corresponding_to_width = static_cast<T>(static_cast<float>(width()) / aspect_ratio);
```

Comment:
possible division by zero.

Reply:
That's just a programming bug; if you provide an aspect ratio of 0 it's impossible to match it. We could VERIFY but that has the same effect. The aspect ratio should be verified beforehand.